src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs

```diff
@@ -384,6 +384,8 @@ public sealed class SourcesCommandConfiguration
         public string Username { get; set; }
         public string Password { get; set; }
         public int Priority { get; set; }
+        public string Certificate { get; set; }
+        public string CertificatePassword { get; set; }

```

Comment:
Yes, the public key from the certificate will be validated, by whatever is doing HTTPS termination, which may or may not be the NuGet server. The certificate validation does not however use password, but the chain of trust and revocation lists. 
Yes, just like there is apikey alongside the username/password, you could have the need for either/both, depending on how the server is implemented.